src/software/ai/fsm_diagram_generator.py

```diff
@@ -121,7 +121,7 @@ def remove_suffix(str, suffix):
             if diagram:
 
                 # Regex match to extract FSM struct name
-                fsm_name = re.findall(r"struct\s+(\w+)\s*\{", fsm)[0]
+                fsm_name = re.findall(r"struct\s+(\w+)\s(?:\s:\s.)?{?", fsm)[0]
```

Comment:
out of curiosity, what was the issue with the regex matching?

Reply:
This tries extracting the FSM name by looking for whatever is after the keyword `struct`, but if we have an FSM that extends another class (e.g. `struct EnemyFreeKickPlayFSM : public DefensePlayBase`) then there's no match. This new regex matches an optional `: ACCESS_MODIFIER PARENT_CLASS_NAME` after the `struct NAME`